### ISSUE TITLE
Scrollbars shouldn't overlap text windows

### ISSUE THREAD

**Issue by rickmcgeer** [LivelyKernel/lively.next#605]
See this screenshot:
![Untitled drawing (4)](https://user-images.githubusercontent.com/656259/210594647-cf8f7217-d9ae-4931-a6ea-99c7b88fde2a.png)

The last line of text in the window is underneath (z-plane) the scrollbar; the text should be above (y-plane) the scrollbar

**Comment by linusha** [comment]
In general - agreed. This problem is even worse for the scrollbar to the right, which might overlap text in the line one is currently editing.

However, due to the way the Textmorph is implemented and the fact that CSS gives us nearly no control over/information about the scrollbar, implementing this is not trivial. Is this a great pain point for you? Or just something you noticed?

---
For future reference: A strategy for implementing this could be making the text-layer smaller for Textmorph in which the scrollbar is visible by something like 15px. Doing this all the time, would probably look very weird; however changing the extent of the text-layer requires a full remeasure of the text-layout, because new lines could wrap now. Showing the scrollbars is currently happening on hover over scrollable textmorphs. This might tank the performance, which would then require us to make further rendering deviations based on some config, whether all this behavior is desired or not. To make it all even worse, this is solely a Linux/Windows problem, as the scrollbars on MacOS are natively much smaller.

**TL;DR:** I agree, that this is not great. I would heavily suggest not fixing this.

**Comment by linusha** [comment]
By the way, assuming this is all about the optical flaw and/or clicking the text being "blocked"  by the scrollbar - the behavior in VS Code (also web based) is the same:

![image](https://user-images.githubusercontent.com/14252419/210770627-1b64718e-cd29-4f34-9bc8-fc0d71714777.png)

Thus, I'll close this. Feel free to reopen if I got something wrong!

**Comment by rickmcgeer** [comment]
This is a minor issue, so feel free to close.  I'm pointing it out because if I don't, you might not be aware of it.  The problem is partly visibility (I can't see the line well) and partly UI/UX (the scrollbar takes mouse/keyboard events I mean for the morph)
The workaround (for me) is just putting blank lines under the last text line...this works well in almost all cases.
